### PR TITLE
Add new firewall rule for app.yaml.

### PR DESCRIPTION
**Pull request by Crell**

### CODE REVIEW

File: src/configuration/app/firewall.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Outbound firewall
+
+In some situations, compliance regulations may require you to limit outbound traffic from your application.  The `firewall` property allows you to do so.
+
+This setting has no impact on inbound requests to your application.  For that, use the environment access control settings in the Management Console.
+
+## Syntax
+
+The `firewall` property defines one or more whitelist entries for outbound requests.  It's basic syntax is as follows:
```
Comment (ralt) [inline src/configuration/app/firewall.md]: Its basic syntax*

File: src/configuration/app/firewall.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+### `ips`
+
+This property is an array of IP addresses in [CIDR notation](https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing).  CIDR allows you to specify a range of IP addresses in a compact format, using a bitmask.  Most commonly the bitmask is 8, 16, or 32 but that is not required.
+
+For example, `1.2.3.4/8` will match any IP address whose first 8 bits match `1.2.3.4`, which corresponds to the first segment.  Therefore it will allow `1.*.*.*`.  In comparison, `1.2.3.4/24` will allow `1.2.3.*`.  A mask of 32 will match only the IP address specified, so to whitelist a single specific IP you must write `1.2.3.4/32`.
+
+[IP Address Guide](https://ipaddressguide.com/cidr) has a useful CIDR format calculator.
+
+This is the only required property.
```
Comment (ralt) [inline src/configuration/app/firewall.md]: Either `ips` or `port` is required.

File: src/configuration/app/firewall.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+That is, all outbound TCP traffic is allowed on all ports (aside from port 25, which is always blocked without exception).  In the majority of cases the default is sufficient for most applications.
+
+## Options
+
+Each firewall rule has three configuration values.
+
+### `protocol`
+
+The protocol value is always `tcp`.  Outbound UDP request are not allowed anyway.  As a result this property can be omitted in virtually every circumstance.
```
Comment (ralt) [inline src/configuration/app/firewall.md]: the default value is "tcp" also, so you don't have to specify it. The "are not allowed anyway" part sounds a bit condescending :)

File: src/configuration/app/firewall.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+If no `ports` property is specified, requests to any port on the specified IP addresses are permitted unless restricted by another firewall rule.
+
+### `ports`
+
+To restrict a rule to only allow requests to certain ports as well, list the ports in this property.  For example, `[80, 443]` will only allow requests to the specified IPs on ports 80 and 443 (typically HTTP and HTTPS, respectively).  Requests to any other port will be blocked.
+
+If not specified, requests to a given IP may be to any port.  Legal values are integers from 1 to 65535.
+
+If no `ips` property is specified, requests to any IP address are permitted on the specified port(s) unless restricted by another firewall rule.
```
Comment (ralt) [inline src/configuration/app/firewall.md]: there's no "unless restricted by another firewall rule", you only allow things.

File: src/configuration/app/firewall.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+For example, `1.2.3.4/8` will match any IP address whose first 8 bits match `1.2.3.4`, which corresponds to the first segment.  Therefore it will allow `1.*.*.*`.  In comparison, `1.2.3.4/24` will allow `1.2.3.*`.  A mask of 32 will match only the IP address specified, so to whitelist a single specific IP you must write `1.2.3.4/32`.
+
+[IP Address Guide](https://ipaddressguide.com/cidr) has a useful CIDR format calculator.
+
+If no `ports` property is specified, requests to any port on the specified IP addresses are permitted unless restricted by another firewall rule.
+
+### `ports`
+
+To restrict a rule to only allow requests to certain ports as well, list the ports in this property.  For example, `[80, 443]` will only allow requests to the specified IPs on ports 80 and 443 (typically HTTP and HTTPS, respectively).  Requests to any other port will be blocked.
```
Comment (ralt) [inline src/configuration/app/firewall.md]: I couldn't say why, maybe I'm tired, but the wording there is hard for me

File: docs/src/configuration/app/firewall.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
+# Outbound firewall
```
Comment (chadwcarlson) [inline docs/src/configuration/app/firewall.md:1]:
```suggestion
---
title: "Outbound firewall"
weight: 15
---
```

Comment (chadwcarlson) [reply docs/src/configuration/app/firewall.md:1]: This will add it to the sidebar.

Comment (chadwcarlson) [reply docs/src/configuration/app/firewall.md:1]: I'm not sure if we want this to show up last in that section, but  that's what this weight will do.

**Pull request merged**